metadetect/metadetect.py

```diff
@@ -115,57 +201,261 @@ def go(self):
 
         if res is not None:
             res = self._add_positions_and_psf(medsifier.cat, res, shear_str)
 
         return res
 
+    def _run_fitter(self, mbobs_list, nonshear_mbobs_list=None):
+        if self['model'] in ['wmom']:
+            return self._run_fitter_mbobs_sep(
+                mbobs_list, nonshear_mbobs_list=nonshear_mbobs_list,
+            )
+        elif self['model'] in ['gauss']:
+            return self._run_fitter_mbobs_comb(
+                mbobs_list, nonshear_mbobs_list=nonshear_mbobs_list,
+            )
+        else:
+            raise RuntimeError(
+                "shear model %s not supported!" % self['model']
+            )
+
+    def _run_fitter_mbobs_comb(self, mbobs_list, nonshear_mbobs_list=None):
+        res = self._fitter.go(mbobs_list)
+        if nonshear_mbobs_list is not None:
+            res_nonshear = self._nonshear_fitter.go(nonshear_mbobs_list)
+            tot_nband = self.nband + self.nonshear_nband
+        else:
+            tot_nband = self.nband
+
+        n = Namer(front=self['model'])
+        if tot_nband > 1:
+            new_dt = [
+                (n("band_flux"), "f8", tot_nband),
+                (n("band_flux_err"), "f8", tot_nband),
+            ]
+        else:
+            new_dt = [
+                (n("band_flux"), "f8"),
+                (n("band_flux_err"), "f8"),
+            ]
+        newres = eu.numpy_util.add_fields(
+            res,
+            new_dt,
+        )
+        if tot_nband > 1:
+            if self.nband > 1:
+                newres[n("band_flux")][:, :self.nband] = res[n('flux')]
+                newres[n("band_flux_err")][:, :self.nband] = res[n('flux_err')]
+            else:
+                newres[n("band_flux")][:, :self.nband] = res[n('flux')].reshape(-1, 1)
+                newres[n("band_flux_err")][:, :self.nband] \
+                    = res[n('flux_err')].reshape(-1, 1)
+        else:
+            newres[n("band_flux")] = res[n('flux')]
+
+        if nonshear_mbobs_list is not None:
+            if self.nonshear_nband > 1:
+                newres[n("band_flux")][:, self.nband:] = res_nonshear[n('flux')]
+                newres[n("band_flux_err")][:, self.nband:] = res_nonshear[n('flux_err')]
+            else:
+                newres[n("band_flux")][:, self.nband:] \
+                    = res_nonshear[n('flux')].reshape(-1, 1)
+                newres[n("band_flux_err")][:, self.nband:] \
+                    = res_nonshear[n('flux_err')].reshape(-1, 1)
+
+        # remove the flux column
+        new_dt = [
+            dt
+            for dt in newres.dtype.descr
+            if dt[0] not in [n("flux"), n("flux_err")]
+        ]
+        final_res = np.zeros(newres.shape[0], dtype=new_dt)
+        for c in final_res.dtype.names:
+            final_res[c] = newres[c]
+
+        return final_res
+
+    def _run_fitter_mbobs_sep(self, mbobs_list, nonshear_mbobs_list=None):
```

Comment:
add doc that this performs a measurement separately on each band, and that for the shear estimator we perform a weighted average across bands